ValheimPlus/GameClasses/Inventory.cs

```diff
@@ -80,4 +80,42 @@ public static void Prefix(string name, ref int w, ref int h)
 			}
 		}
 	}
+
+	/// <summary>
+	/// When merging another inventory, try to merge items with existing stacks.
+	/// </summary>
+	[HarmonyPatch(typeof(Inventory), "MoveAll")]
+	public static class Inventory_MoveAll_Patch
+	{
+		private static void Prefix(ref Inventory __instance, ref Inventory fromInventory)
+		{
+			if (Configuration.Current.Inventory.IsEnabled && Configuration.Current.Inventory.mergeWithExistingStacks)
+			{
+				List<ItemDrop.ItemData> list = new List<ItemDrop.ItemData>(fromInventory.GetAllItems());
+				foreach (ItemDrop.ItemData otherItem in list)
+				{
+					if (otherItem.m_shared.m_maxStackSize > 1)
+					{
+						foreach (ItemDrop.ItemData myItem in __instance.m_inventory)
+						{
+							if (myItem.m_shared.m_name == otherItem.m_shared.m_name && myItem.m_quality == otherItem.m_quality)
```

Comment:
I would think most stackable items don't have a quality, but I suppose that doesn't account for modded items.  I haven't yet learned enough to mod valheim yet but are there other attributes we might need to check for items to ensure we don't stack things that have other attributes?  Though how do other mods deal with stackable items that have qualities and such?

Reply:
Yes it doesn't make sense for stackable items to have a quality but the original decompiled code also has this check so it should stay.

I think in the past when minecraft had a limited number of item ids the damage field was abused to cram many items into the same item id. As long as the original code supports this abuse we should support it too.